<?xml version="1.0"?>
<sdf version="1.4">
  <world name="default">
    <include>
      <uri>model://ground_plane</uri>
    </include>

    <include>
      <uri>model://sun</uri>
    </include>
    <plugin name="wirelessSimulator" filename="libwirelessSimulator.so"/>
    <!-- Drone 1 -->
    
    <model name="drone1">
      <pose>0 0 0.5 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <inertial>
          <pose>0 0 0 0 0 0</pose>
        </inertial>
        <collision name="collision">
          <geometry>
            <box>
              <size>0.5 0.5 0.5</size>
            </box>
          </geometry>
        </collision>

        <visual name="visual">
          <geometry>
            <box>
              <size>0.5 0.5 0.5</size>
            </box>
          </geometry>

          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Green</name>
            </script>
          </material>
        </visual>
      </link>
      <plugin name="meshnetworkCom" filename="libmeshnetworkCom.so"/>
      <plugin name="droneEngine" filename="libdroneEngine.so">
        <DroneID>2</DroneID>
      </plugin>
    </model>

    <!-- Drone 2 -->
    <!--
    <model name="drone2">
      <pose>0 0 0.5 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <inertial>
          <pose>0 0 0 0 0 0</pose>
        </inertial>
        <collision name="collision">
          <geometry>
            <box>
              <size>0.5 0.5 0.5</size>
            </box>
          </geometry>
        </collision>

        <visual name="visual">
          <geometry>
            <box>
              <size>0.5 0.5 0.5</size>
            </box>
          </geometry>

          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Green</name>
            </script>
          </material>
        </visual>
      </link>
      <plugin name="droneEngine" filename="libdroneEngine.so">
        <DroneID>1</DroneID>
      </plugin>
      <plugin name="meshnetworkCom" filename="libmeshnetworkCom.so"/>
    </model>

    <!-- Drone 2 -->
    <model name="gateway">
      <pose>0 0 0.5 0 0 0</pose>
       <static>true</static>
      <link name="link">
        <inertial>
          <pose>0 0 0 0 0 0</pose>
          <mass>100</mass>
          <inertia>
            <ixx>3.33341666667e+3</ixx>
            <ixy>0.0</ixy>
            <ixz>0.0</ixz>
            <iyy>3.33341666667e+3</iyy>
            <iyz>0.0</iyz>
            <izz>6.66666666667e+3</izz>
          </inertia>
        </inertial>
        <collision name="collision">
          <geometry>
            <box>
              <size>0.5 0.5 0.5</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>0</mu>
                <mu2>0</mu2>
              </ode>
            </friction>
          </surface>
        </collision>

        <visual name="visual">
          <geometry>
            <box>
              <size>0.5 0.5 0.5</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Red</name>
            </script>
          </material>
        </visual>
      </link>
      <plugin name="droneEngine" filename="libdroneEngine.so">
        <DroneID>2</DroneID>
      </plugin>
      <plugin name="meshnetworkGateway" filename="libmeshnetworkGateway.so"/>
    </model>

  </world>
</sdf>
